spec/models/factory_girl_spec.rb

```diff
@@ -0,0 +1,21 @@
+describe 'FactoryGirl' do
+  local_models = Dir.glob('app/models/*.rb').map { |file| File.basename(file, '.rb') }
+
+  ApplicationRecord.subclasses.each do |klass|
+    model = klass.name.underscore
+    next unless local_models.include?(model) # Skip non-locally defined models
+
+    context klass.name do
+      it 'factory can be built' do
+        begin
+          FactoryGirl.build(model.to_sym)
+        rescue ArgumentError # This occurs when the factory is not defined
+          raise "There's no factory defined for the model '#{klass}'"
+        rescue # This occurs when the factory is defined but it can't be built without extra context
```

Comment:
Can this even happen?  `.build` doesn't call `.save`, and thus won't run any validations.  This is like calling `.new`, which should *always* work.

Reply:
yup, this fails sometimes

Reply:
Wouldn't this be better if instead of the `begin / rescue` logic...
```ruby
local_model_files = Dir.glob('app/models/*.rb').map { |file| File.basename(file, '.rb').to_sym }
all_models = ApplicationRecord.descendants.collect { |d| d.name.underscore.to_sym }
expected_factories = all_models & local_model_files
registered_factories = FactoryGirl.factories.collect(&:name)
expect(registered_factories).to include(expected_factories)
```
Otherwise you could have multiple undefined factories and you'll have to fix and re-run in a loop until all are defined, right?  This way, you should get a list of all the missing factories.  Also, you don't need to instantiate each one and deal with this awkward rescue situation (and potentially saving memory / cpu).

Reply:
@bdunne I agree, but I'm a bit afraid to drop the *test case for each model* logic as it's more descriptive. I feel that there's a chance that some people will *fix the test* instead of the issue.